Makefile

```diff
@@ -147,17 +177,38 @@ rpm: $(CR_BIN) $(PKG_INFO)
 .PHONY: docker-rpm
 docker-rpm:
 	$(call print_msg, Running ClusterRunner Docker RPM builder... )
-	$(eval TAG := productivity/clusterrunner)
+	docker build -t $(DOCKER_TAG) -f Dockerfile .
 	mkdir -p $(DIST_DIR)
-	docker build -t $(TAG) -f Dockerfile .
-	@# Docker cp does not support globing, so detect the path to the RPM file.
-	$(eval RPM_PATH := $(shell docker run $(TAG) sh -c "ls /root/$(DIST_DIR)/*.rpm"))
-	@# Docker "run" must be called before the next steps.
-	$(eval CONTAINER_ID := $(shell docker ps -alq))
-	docker cp $(CONTAINER_ID):$(RPM_PATH) dist/
+	@# Docker cp does not support globing, so the path to the RPM file must be
+	@# detected with a query. The order of the commands are important and they
+	@# must all be run in the same "shell" for the variables to be available to
+	@# the final "docker cp" command.
+	DOCKER_RPM_PATH=$$(docker run $(DOCKER_TAG) sh -c "ls /root/$(DIST_DIR)/*.rpm") && \
+	CONTAINER_ID=$$(docker ps -alq) && \
+	docker cp $$CONTAINER_ID:$$DOCKER_RPM_PATH $(DIST_DIR)
```

Comment:
`docker rm` the container after you grab the rpm out of it

Reply:
👍